openpype/pipeline/create/creator_plugins.py

```diff
@@ -611,6 +614,52 @@ def get_pre_create_attr_defs(self):
         """
         return self.pre_create_attr_defs
 
+    def get_custom_staging_dir_data(self, instance, task_type=None):
+        """Custom staging directory data.
+
+        Arguments:
+            instance (CreatedInstance): Instance for which we want to get
+                staging dir.
+            task_type (str): Task type. If not set it will be queried from
+        Returns:
+            Tuple[Any, Any]: Tuple of staging dir and is_persistent or None
+        Raises:
+            ValueError - if misconfigured template should be used
+        """
+        from openpype.pipeline import get_custom_staging_dir_info
+
+        task_name = instance["task"]
+        task_type = task_type or self.get_task_type(task_name)
+
+        return get_custom_staging_dir_info(
+            self.project_name,
+            self.host_name,
+            self.family,
+            task_name,
+            task_type,
+            instance["subset"],
+            project_settings=self.project_settings,
+            anatomy=self.project_anatomy, log=self.log
+        )
+
+    def get_task_type(self, task_name):
+        """Get task type from task name.
+
+        Args:
+            task_name (str): Task name.
+
+        Returns:
+            str: Task type.
+        """
+        create_context = self.create_context
+        project_name = create_context.get_current_project_name()
+        asset_name = create_context.get_current_asset_name()
+        task_name = create_context.get_current_task_name()
+
+        asset_doc = get_asset_by_name(project_name, asset_name)
+
+        return get_task_type(asset_doc, task_name)
```

Comment:
If `get_task_type` takes the `asset_doc` shouldn't the call to `self.get_task_type` also actually use the asset from the instance instead of the global create context?

Reply:
Good point. Need to look into this, since it is really optional in TrayPublisher.

Reply:
Is the question related to editorial case? I presume we could face a problem there, but still it feels like you would want to use transient data linked to editorial task (and asset parent) rather than to each individual shot instances. Or is there another case you are having in mind?

Reply:
@BigRoy how about to do it this way? https://github.com/ynput/OpenPype/pull/5207/commits/a31ce9ad14713cc70cdb17ed2739fd30d9c5cbf3

Reply:
Actually yes, because to me this actually goes to show that `get_task_type` is not really needed as a method to have it on the `Creator` plugin to begin with since it's a very thin wrapper to `openpype.pipeline.template_data.get_task_type`. Or better yet, why do we need `get_staging_dir_profile` on the `Creator` plugin? Is it even used - it doesn't appear to be used in this PR?

Reply:
Yeah, this is preparation for creators. Those will be using `get_staging_dir_profile` or better `apply_staging_dir` selectively. The idea is to call the `apply_staging_dir` method from particular creator and then have collected per host which will add it to stagingDir instance.data.

Reply:
> Actually yes, because to me this actually goes to show that get_task_type is not really needed as a method to have it on the Creator plugin to begin with since it's a very thin wrapper to openpype.pipeline.template_data.get_task_type

You are right again. I have found better solution. Thanks for suggesting it ;)